### PR TITLE
Add removing and adding of transition edges

### PR DESCRIPTION
**Pull request by DOBEN**
## Purpose

Addresses https://github.com/Concordium/concordium-dapp-examples/issues/75

To simplify the use of this demo. We want to enable productTeam to easily add themself as a privileged role into the state machine.

## Changes
Smart contract changes:
- Add functions to update (add and remove) state transition edges in the smart contract.

Front end changes:
- Add front end page for transition edge management by admin.
- Add link to hosted front end in Read.me file.
- Display front-end version and link to source code in the nav bar.
- Add /Fix `eslint` errors/warnings

### CODE REVIEW

File: trackAndTrace/frontend/src/components/Explorer.tsx [hunk 1]
```diff
@@ -103,11 +103,11 @@ export function Explorer() {
             throw Error(`'itemID' input field is undefined`);
         }
 
-        // Clear table.
-        const tableBody = document.getElementById('table');
-        if (tableBody) {
-            tableBody.innerHTML = '';
-        }
+        // // Clear table.
```
Comment (abizjak) [inline trackAndTrace/frontend/src/components/Explorer.tsx:106]: Presumably this should be removed?

File: trackAndTrace/frontend/vite.config.ts [hunk 2]
```diff
@@ -84,6 +84,15 @@ const viteConfig: UserConfig = {
...
         },
     },
+    define: {
+        global: 'globalThis',
+        'process.env': {
+            TRACK_AND_TRACE_NETWORK: process.env.TRACK_AND_TRACE_NETWORK,
+            TRACK_AND_TRACE_SPONSORED_BACKEND_API: process.env.TRACK_AND_TRACE_SPONSORED_BACKEND_API,
+            TRACK_AND_TRACE_NODE: process.env.TRACK_AND_TRACE_NODE,
+            TRACK_AND_TRACE_CONTRACT_ADDRESS: process.env.TRACK_AND_TRACE_CONTRACT_ADDRESS,
+        },
```
Comment (abizjak) [inline trackAndTrace/frontend/vite.config.ts:94]: Why is this necessary?

File: trackAndTrace/smart-contract/src/lib.rs [hunk 3]
```diff
@@ -247,6 +238,8 @@ pub enum CustomContractError {
     WrongEntryPoint, // -14
     /// Failed signature verification: Signature is expired.
     Expired, // -15
+    /// Update of state machine was unsuccessful.
+    UnSuccessful, // -16
```
Comment (abizjak) [inline trackAndTrace/smart-contract/src/lib.rs:242]:
```suggestion
    Unsuccessful, // -16
```

File: trackAndTrace/smart-contract/src/lib.rs [hunk 4]
```diff
@@ -326,6 +319,24 @@ impl<S: HasStateApi> State<S> {
         targets.insert(to)
     }
 
+    /// Remove a transition. Return if the transition was removed successfully.
```
Comment (abizjak) [inline trackAndTrace/smart-contract/src/lib.rs:322]:
```suggestion
    /// Remove a transition. Return if the transition was present.
```

File: trackAndTrace/smart-contract/src/lib.rs [hunk 5]
```diff
@@ -326,6 +319,24 @@ impl<S: HasStateApi> State<S> {
...
+        from: Status,
+        address: AccountAddress,
+        to: Status,
+    ) -> bool {
+        let mut transition = self
+            .transitions
+            .entry(from)
+            .or_insert_with(|| StatusTransitions {
+                transitions: builder.new_map(),
+            });
```
Comment (abizjak) [inline trackAndTrace/smart-contract/src/lib.rs:335]: This seems needless. If the entry is not there should you not just return false immediately?


```rust
        let Some(mut transition) = self
            .transitions
            .get_mut(from) else { return false; };
```
???

File: trackAndTrace/smart-contract/src/lib.rs [hunk 6]
```diff
@@ -636,6 +647,84 @@ fn change_item_status(
...
+        }
+        Update::Remove => {
+            let success = state.remove(
+                state_builder,
+                params.from_status,
+                params.address,
+                params.to_status,
+            );
+
+            ensure!(success, CustomContractError::UnSuccessful);
```
Comment (abizjak) [inline trackAndTrace/smart-contract/src/lib.rs:721]: In principle we don't need to do this check. If this returns false it just means there was nothing to remove. So nothing was changed.

File: trackAndTrace/frontend/vite.config.ts (hunk 2 above)
Comment (DOBEN) [reply trackAndTrace/frontend/vite.config.ts:94]: It is necessary to run the `front-end` locally without the rest of the components. If you follow the read.me, you get an error trying to run `yarn dev` without these definitions.

File: trackAndTrace/smart-contract/src/lib.rs (hunk 6 above)
Comment (DOBEN) [reply trackAndTrace/smart-contract/src/lib.rs:721]: I felt it would be better to do it that way, since `productTeam` can now figure out the current state of the `stateMachine` by trying to add/remove an edge via the front end (and observe if the front-end returns an error or not).

**Review by Bargsteen — APPROVED** [review]
LGTM. Works as expected :tada:
